lib/autoproj/ros_package_manifest.rb

```diff
@@ -23,10 +23,13 @@ class Loader < PackageManifest::Loader
                              buildtool_depend buildtool_export_depend
                              exec_depend test_depend run_depend doc_depend].to_set.freeze
 
-            def initialize(path, manifest)
+            def initialize(path, manifest, condition_context: {})
                 super
-                @env = manifest.package.ws.env
-                @condition_parser = RosConditionParser.new(@env)
+                @condition_parser = RosConditionParser.new do |var|
+                    Autoproj.expand(var, condition_context)
+                rescue StandardError
```

Comment:
Since you removed this from the condition parser could you please add a test case to `test_ros_package_manifest` that ensures unset variables evaluate to empty strings in ROS manifests as expected by ROS packages? You've added the expander to the condition parser's test suite and you are testing it there at the same time which is kinda pointless.

Reply:
> You've added the expander to the condition parser's test suite and you are testing it there at the same time which is kinda pointless.

You are right, moved that specific test up to test_ros_package_manifest, testing that behavior through the manifest functions.

Reply:
Thanks